packages/components/src/components/checkbox/checkbox.stories.ts

```diff
@@ -32,318 +35,83 @@ export const Default = {
+ * Use the `required` attribute to mark the element as required. This can be used for form validation purposes.
  */
-export const Parts = {
-  parameters: {
-    controls: {
-      exclude: [
-        'base',
-        'control',
-        'control--unchecked',
-        'control--checked',
-        'checked-icon',
-        'control--indeterminate',
-        'indeterminate-icon',
-        'label',
-        'title',
-        'name',
-        'value',
-        'size',
-        'disabled',
-        'checked',
-        'indeterminate',
-        'form',
-        'required',
-        'default'
-      ]
-    }
-  },
-  render: (args: any) => {
-    return generateTemplate({
-      axis: {
-        y: {
-          type: 'template',
-          name: 'sd-checkbox::part(...){outline: solid 2px red}',
-          values: [
-            'base',
-            'control',
-            'control--unchecked',
-            'control--checked',
-            'checked-icon',
-            'control--indeterminate',
-            'indeterminate-icon',
-            'label'
-          ].map(part => {
-            return {
-              title: part,
-              value: `
-                <style>
-                    #part-${part} sd-checkbox::part(${part}){outline: solid 2px red};
-                    .hidden {display: none}
-                </style>
-                <div id="part-${part}">${checkboxTemplate(part)}</div>
-                <div class="hidden">%TEMPLATE%</div>
-              `
-            };
-          })
-        }
-      },
-      args
-    });
-  }
-};
 
-const checkboxTemplate = (part: string) => {
-  switch (part) {
-    case 'control--checked':
-      return `<sd-checkbox checked>Default Slot</sd-checkbox>`;
-    case 'checked-icon':
-      return `<sd-checkbox checked>Default Slot</sd-checkbox>`;
-    case 'control--indeterminate':
-      return `<sd-checkbox indeterminate>Default Slot</sd-checkbox>`;
-    case 'indeterminate-icon':
-      return `<sd-checkbox indeterminate>Default Slot</sd-checkbox>`;
-    case 'form-control-error-text':
-      return `<sd-checkbox error-text="Error message" invalid>Default Slot</sd-checkbox>`;
-    default:
-      return `<sd-checkbox>Default Slot</sd-checkbox>`;
-  }
+export const Required = {
+  name: 'Required',
+  render: () => html`
+    <div class="flex gap-12">
+      <sd-checkbox required>Required</sd-checkbox>
+    </div>
+  `
 };
 
 /**
- * 1. You can use the `setCustomValidity` method to set a custom validation message. This will override any native validation messages.
- * 2. Set an empty string to clear the custom validity and make the input valid.
- * 3. To show the validation message, call the `reportValidity` method. Originally this would show a native validation bubble, but we show the error messages inline.
+ * Test invalid state inside a form.
+ *
+ * For an invalid checkbox an error-text underneath (or if used in a group underneath the checkbox-group) is mandatory.
```

Comment:
Invalid in Figma isn't marked as required. @coraliefeil is that right?
Also the Label is "Invalid" not Checkbox.

<img width="638" alt="image" src="https://github.com/user-attachments/assets/ef37ba93-fe49-4c37-b6ea-a01e9ceaaa80">

Reply:
@coraliefeil @MartaPintoTeixeira we need `required` in code due to form validation. Please check if it's worth adding in figma as @yoezlem mentioned.